### PR TITLE
Translations update from Weblate

### PR DESCRIPTION
**Pull request by LXQtBot**
Translations update from [Weblate](https://translate.lxqt-project.org/projects/testing/labwc/) for Testing/Labwc.

### CODE REVIEW

**Comment by Consolatis** [comment]
@stefonarch Thanks for integrating labwc into the lxqt weblate instance!

It seems to operate partly an a slightly outdated labwc version (before #1343 was merged).
Example: msgid `ToggleOmnipresent` in e34ea93c3275f7e3270a4fc9ca4db2ec4f0f3de7 vs `Always on Visible Workspace` in e9987445f888efbc816119837e81385732f194e3

We should likely also squash mass commit PRs like this one but can do that when merging.

CC @johanmalm @01micko

**Comment by 01micko** [comment]
IDK if it supposed to fill in the `msgstr=""` entries but it seems all are empty. (At least the half dozen or more I checked).

@stefonarch does the bot understand `gettext`? It seems QT tanslations uses xml style `.ts` ([translation source](https://docs.fileformat.com/web/ts/#references)?) files for translation.

**Comment by stefonarch** [comment]
I made just a quick one-minute test yesterday (not visible) but actually didn't configure the bot for PRs, so this here shouldn't even exist. Mmaybe @yan12125 did it? There was also a plugin installed which created all the empty language files and a plugin to squash commits can be installed ( did it here now).
Anyway if this is fine with you I've to  reconfigure it outside of the "testing" instance and more testing.

Weblate can handle gettext well, just I've never used it, I see several plugins:

https://docs.weblate.org/en/weblate-4.2.1/admin/addons.html#addon-weblate-gettext-customize
https://docs.weblate.org/en/weblate-4.2.1/admin/addons.html#addon-weblate-gettext-msgmerge

>can we still accept translation PRs and weblate will then just update its internal state?

This could lead to merge conflicts if the same file get's edited on both sides, but I think with few translations the risk is small.

**Comment by stefonarch** [comment]
Installed some plugins and updated the italian translation now.

File: po/de.po [hunk 1]
```diff
@@ -17,46 +17,52 @@ msgstr ""
...
+#: src/menu/menu.c:722
 msgid "Decorations"
 msgstr "Dekorationen"
 
-#: src/menu/menu.c:657
-msgid "Always On Top"
+#: src/menu/menu.c:724
+#, fuzzy
+#| msgid "Always On Top"
+msgid "Always on Top"
```
Comment (Consolatis) [inline po/de.po:47]: Oh, it appears we have used `On` in the `.po` files but `on` (lowercase) as actual msgid and in the C sources.
We should likely fix that.

Comment (johanmalm) [reply po/de.po:47]: Good catch!

**Comment by stefonarch** [comment]
I see now that also `src/menu/` has to be added. Will do more testing this evening.

screen_area_gio_08:13:04_

File: po/de.po (hunk 1 above)
Comment (01micko) [reply po/de.po:47]: > Oh, it appears we have used `On` in the `.po` files but `on` (lowercase) as actual msgid and in the C sources. We should likely fix that before 0.7.0.

Looks like the bot has caught that! (see latest force push)

**Comment by yan12125** [comment]
> Maybe @yan12125 did it?

Nope, you overestimate my speed!

**Comment by Consolatis** [comment]
> EDIT: it works, but is visible on weblate only after merge.

Well, currently https://translate.lxqt-project.org/projects/testing/labwc/ just tells me "Page not found", maybe a permission issue?
Also, will it keep using the `/testing/` prefix?

JFYI, I see that the bot now added an empty `po/pt.po`. Having those around as templates doesn't sound that bad to me so I don't really mind. Just wanted to make sure it was not added by some mistake somewhere.

**Comment by stefonarch** [comment]
As I said, I've to move it out from testing (which is not visible for non admins), so the /testing prefix will disappear, will do so later today, it looks working to me.

I added for testing purpose a pt file, to test if LINGUAS will be updated, good to know that this is no issue also if empty. Thought of adding the most common language files, not all translators have the permits to do so,

The bot just creates or updates the PR, all files get created/modified by weblate.

**Comment by stefonarch** [comment]
> > Maybe @yan12125 did it?
> 
> Nope, you overestimate my speed!

So who forked the labwc repo today? Could be that it will be forked automatically... good to know, I didn't intend to create this PR.

**Comment by yan12125** [comment]
> Could be that it will be forked automatically

Sounds likely. Only LXQtBot can fork a repo into https://github.com/LXQtBot

File: po/de.po (hunk 1 above)
Comment (johanmalm) [reply po/de.po:47]: Looks like the bot has fixed this now :smile:

**Comment by stefonarch** [comment]
https://translate.lxqt-project.org/projects/labwc/labwc/ is ready and made already a PR I see.
Should be fine merge the other and close this here but check yourself.

**Comment by johanmalm** [comment]
Thanks. I'll close this one then.
Italian and Swedish are looking particularly good :smile:

**Comment by stefonarch** [comment]
Weblate marked now as "needs editing" the translations for changed source for `Always O/on Top`.
I can bulk edit but not sure if I will end up as "last translator".
https://github.com/labwc/labwc/blob/9ea6104fce284104ad0e2e93a50521cacc54f147/po/de.po#L46

**Comment by johanmalm** [comment]
Oh, sorry. What should I do?

**Comment by stefonarch** [comment]
Not sure yet  how to handle that, it has no effect  as translation exists. For some reason also the italian file didn't update to 12 strings. There is another PR hanging now that will get force-pushes. let's see.

**Comment by johanmalm** [comment]
Shall I merge #1349?

**Comment by stefonarch** [comment]
> Shall I merge #1349?

It's as I thought, bulk removing of the "needs editing" flag makes me "last translator", so I did reset. This PR is ready for italian now.

**Comment by stefonarch** [comment]
Atm there is a plugin active which updates the` .po` files automatically on changes in `labwc.pot`, should I keep that active or is this done otherwise? Not urgent as there won't be many changes I guess.

**Comment by Consolatis** [comment]
> Atm there is a plugin active which updates the` .po` files automatically on changes in `labwc.pot`, should I keep that active or is this done otherwise? Not urgent as there won't be many changes I guess.

If we can automate this that would be great. So I'd vote for keeping it active.

Added a suggestion for the missing German translation.
Also thanks again for sharing part of the lxqt infrastructure, it is really appreciated :)

**Comment by johanmalm** [comment]
Yes, agree with active and agree with appreciation :smile:

**Comment by stefonarch** [comment]
It's nice we can something back to labwc ;)

To remove the "needs editing" state it should work removing all the 
```
#, fuzzy
#| msgid "Always On Top"
```
lines from the `.po` files, but this should be done with weblate in sync, with no commits pending there.

**Pull request closed without merging**